Testscripts/Windows/PERF-XDP-NTTTCP.ps1

```diff
@@ -0,0 +1,265 @@
+        $constantsFile = "$LogDir\constants.sh"
+        Set-Content -Value "#Generated by Azure Automation." -Path $constantsFile
+        Add-Content -Value "ip=$($receiverVMData.InternalIP)" -Path $constantsFile
+        Add-Content -Value "client=$($receiverVMData.InternalIP)" -Path $constantsFile
+        # Giving server IP as secondary to run NTTTCP on secondary NIC.
+        Add-Content -Value "server=$($senderVMData.SecondInternalIP)" -Path $constantsFile
+        Add-Content -Value "nicName=$iFaceName" -Path $constantsFile
+        foreach ($param in $currentTestData.TestParameters.param) {
+            Add-Content -Value "$param" -Path $constantsFile
+        }
+        Write-LogInfo "constants.sh created successfully..."
+        Write-LogInfo (Get-Content -Path $constantsFile)
+        # Start XDP Installation
+        $installXDPCommand = @"
+bash ./XDPDumpSetup.sh 2>&1 > ~/xdpConsoleLogs.txt
+. utils.sh
+collect_VM_properties
+"@
+        Set-Content "$LogDir\StartXDPSetup.sh" $installXDPCommand
+        Copy-RemoteFiles -uploadTo $receiverVMData.PublicIP -port $receiverVMData.SSHPort `
+            -files "$constantsFile,$LogDir\StartXDPSetup.sh" `
+            -username $user -password $password -upload -runAsSudo
+        Copy-RemoteFiles -uploadTo $senderVMData.PublicIP -port $senderVMData.SSHPort `
+            -files "$constantsFile" `
+            -username $user -password $password -upload -runAsSudo
+        $testJob = Run-LinuxCmd -ip $receiverVMData.PublicIP -port $receiverVMData.SSHPort `
+            -username $user -password $password -command "bash ./StartXDPSetup.sh" `
+            -RunInBackground -runAsSudo
+        # Terminate process if ran more than 5 mins
+        # TODO: Check max installation time for other distros when added
+        $timer = 0
+        while ($testJob -and ((Get-Job -Id $testJob).State -eq "Running")) {
+            $currentStatus = Run-LinuxCmd -ip $receiverVMData.PublicIP -port $receiverVMData.SSHPort `
+                -username $user -password $password -command "tail -2 ~/xdpConsoleLogs.txt | head -1" -runAsSudo
+            Write-LogInfo "Current Test Status: $currentStatus"
+            Wait-Time -seconds 20
+            $timer += 1
+            if ($timer -gt 15) {
+                Throw "XDPSetup did not stop after 5 mins. Please check xdpConsoleLogs."
+            }
+        }
+
+        $currentState = Run-LinuxCmd -ip $receiverVMData.PublicIP -port $receiverVMData.SSHPort `
+            -username $user -password $password -command "cat ~/state.txt" -runAsSudo
+        if ($currentState -imatch "Completed") {
+            # Start PERF test without XDP
+            $ResultDir = "$LogDir\WithoutXDP"
+            New-Item -Path $ResultDir -ItemType Directory -Force | Out-Null
+            Run_NTTTCP_PERF $ResultDir
+
+            # Start XDPDump on client
+            $xdp_command = "cd ~/bpf-samples/xdpdump && ./xdpdump -i $iFaceName > ~/xdpdumpoutPERF.txt"

```

Comment:
please add turn off lro part, thanks!

Reply:
It will be handled by XDPDumpsetup.sh. We don't need to turn it off again as we are not rebooting or anything.
What do you think?